### PR TITLE
Add ability to set the requested execution level for UAC

### PR DESCRIPTION
**Pull request by dmascord**
As per http://layer0.authentise.com/electron-and-uac-on-windows.html, add the ability for rcedit.exe to modify execution level in the Manifest.

rcedit.exe modifications are present in https://github.com/dmascord/rcedit, and already submitted for review.

### CODE REVIEW

File: lib/rcedit.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var path = require('path')
 var spawn = require('child_process').spawn
 
 var pairSettings = ['version-string']
-var singleSettings = ['file-version', 'product-version', 'icon']
+var singleSettings = ['file-version', 'product-version', 'icon', 'requested-executation-level']
```
Comment (malept) [inline lib/rcedit.js]: `execution`, not `executation`.

**Comment by malept** [comment]
Also, is it possible to add a test?

File: lib/rcedit.js (hunk 1 above)
Comment (dmascord) [inline lib/rcedit.js]: good catch!

**Comment by dmascord** [comment]
In regards to tests, we can easily enough write a string into the manifest, and then use grep to test that it wrote to it, but testing whether Windows accepts the file might be tough, do you have tests that execute wine to see whether the test binary executes after the modification ?  (Ie, one issue that I noticed is that the padding can get in the way of the "side load configuration test" that Windows does during execution)

**Comment by malept** [comment]
I was just wondering if we can use something like how we're using `rcinfo` to test the other parts of rcedit.

**Comment by dmascord** [comment]
Tests added, unfortunately rcinfo doesn't read the manifest, but fortunately, we can use textgrep on the file contents.

File: test/rcedit-test.js [hunk 2]
```diff
@@ -107,6 +108,24 @@ describe('rcedit(exePath, options, callback)', function () {
     })
   })
 
+  it('supports setting requestedExecutionLevel to requireAdministrator', function (done) {
+    var options = {
+      'requested-execution-level': 'requireAdministrator'
+    }
+
+    rcedit(exePath, options, function (error) {
```
Comment (malept) [inline test/rcedit-test.js]: Before you run rcedit, you should assert that the same grep that's run after running `rcedit` fails.

**Comment by malept** [comment]
> Tests added, unfortunately rcinfo doesn't read the manifest, but fortunately, we can use textgrep on the file contents.

I take it there's no node module that can read an exe manifest...

**Comment by dmascord** [comment]
Not that I have found... rcinfo could be extended...

File: test/fixtures/electron.manifest [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      <!-- Windows 10 -->
+      <supportedOS Id="{8e0f7a12-bfb3-4fe8-b9a5-48fd50a15a9a}"/>
+      <!-- Windows 8.1 -->
+      <supportedOS Id="{1f676c76-80e1-4239-95bb-83d0f6d0da78}"/>
+      <!-- Windows 8 -->
+      <supportedOS Id="{4a2f28e3-53b9-4441-ba9c-d69d4a4a6e38}"/>
+      <!-- Windows 7 -->
+      <supportedOS Id="{35138b9a-5d96-4fbd-8e2d-a2440225f93a}"/>
+      <!-- Windows Vista -->
+      <supportedOS Id="{e2011457-1546-43c5-a5fe-008deee3d3f0}"/>
```
Comment (malept) [inline test/fixtures/electron.manifest]: Er, what? 😵 

_(I know this is a fixture but...)_

Comment (kevinsawicki) [inline test/fixtures/electron.manifest]: Good point, let me nuke most of the fields so it is more intention-revealing.

**Comment by kevinsawicki** [comment]
@malept any other feedback on this one? 👀

File: lib/rcedit.js [hunk 4]
```diff
@@ -26,6 +27,13 @@ module.exports = function (exe, options, callback) {
     }
   })
 
+  noPrefixSettings.forEach(function (name) {
+    if (options[name] != null) {
```
Comment (malept) [inline lib/rcedit.js]: `!==` ?

Comment (kevinsawicki) [inline lib/rcedit.js]: Other similar blocks accessing `options` in this file are using `!=` and it seems good to guard against both `undefined` and `null`.

Comment (malept) [inline lib/rcedit.js]: Fair enough.

**Comment by kevinsawicki** [comment]
Thanks for this @dmascord 👍 

Published in `0.9.0`

**Pull request merged**